sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala

```diff
@@ -62,6 +66,14 @@ object CommandUtils extends Logging {
       if (isNewStats) {
         val newStats = CatalogStatistics(sizeInBytes = newSize)
         catalog.alterTableStats(table.identifier, Some(newStats))
+
+        if (withAutoPartitionStats) {
+          if (partitionSpec.nonEmpty) {
+            AnalyzePartitionCommand(table.identifier, partitionSpec, false).run(sparkSession)
+          } else if (table.partitionColumnNames.nonEmpty) {
+            AnalyzePartitionCommand(table.identifier, partitionSpec, false).run(sparkSession)
```

Comment:
I'm not sure whether we want to always do full scan in the partitions to get number of rows - it could be rather expensive?

Reply:
Right. Would you recommend to make a new config to enable/disable full scan?

The doc says `spark.sql.statistics.size.autoUpdate.enabled` can be expensive anyway, so I chose to do full can always for the first review...
https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L2323

Reply:
I think it's better to disable the full scan for now, since we don't do that either for table stats. We can add the option later if there is a need.

Reply:
updated

Reply:
It turned out that in order to support the dynamic cases, it needs to be `noScan = false`